### PR TITLE
Confidence of rule

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Add confidence for rule which may be used in Candidate
- Fix url password detection in sources with quotes
- Sphinx documentation was updated

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

**Review by kmnls — DISMISSED** [review]
Agree

**Review by csh519 — CHANGES_REQUESTED** [review]
Please check my mail.

File: credsweeper/rules/config.yaml [hunk 1]
```diff
@@ -250,6 +254,7 @@
 
 - name: Google OAuth Access Token
   severity: high
+  confidence: strong
```
Comment (babenek) [inline credsweeper/rules/config.yaml:257]:
```suggestion
```

File: credsweeper/rules/config.yaml [hunk 2]
```diff
@@ -260,6 +265,7 @@
 
 - name: Heroku API Key
   severity: high
+  confidence: strong
```
Comment (babenek) [inline credsweeper/rules/config.yaml:268]:
```suggestion
```

**Review by babenek — COMMENTED** [review]
@csh519 can you apply?

**Comment by csh519** [comment]
Please add confidence level to all rules on `rules/config.yaml` file.

**Review by csh519 — DISMISSED** [review]
Thank you for adding `Confidence` field.
All `Confidence` levels are discussed internally.

LGTM :+1:

**Review by Yullia — DISMISSED** [review]

**Review by kmnls — APPROVED** [review]

File: credsweeper/common/constants.py [hunk 3]
```diff
@@ -56,6 +56,31 @@ def get(severity: Union[str, "Severity"]) -> Optional["Severity"]:
...
+
+    def __lt__(self, other) -> bool:
+        if self == Confidence.WEAK:
+            return other is not Confidence.WEAK
+        elif self == Confidence.MODERATE:
+            return other is Confidence.STRONG
+        return False
+
+    @staticmethod
+    def get(severity: Union[str, "Confidence"]) -> Optional["Confidence"]:
```
Comment (xDizzix) [inline credsweeper/common/constants.py:73]:
```suggestion
    def get(confidence: Union[str, "Confidence"]) -> Optional["Confidence"]:
```

File: credsweeper/common/constants.py [hunk 4]
```diff
@@ -56,6 +56,31 @@ def get(severity: Union[str, "Severity"]) -> Optional["Severity"]:
...
+        """returns Severity value from string or None"""
+        if isinstance(severity, Confidence):
+            return severity
+        if isinstance(severity, str):
+            value = getattr(Confidence, severity.strip().upper(), None)
```
Comment (xDizzix) [inline credsweeper/common/constants.py:78]:
```suggestion
        if isinstance(confidence, Confidence):
            return confidence
        if isinstance(confidence, str):
            value = getattr(Confidence, confidence.strip().upper(), None)
```

**Review by xDizzix — APPROVED** [review]
+

**Review by iuriimet — APPROVED** [review]

**Pull request merged**